src/components/StateProvider/StateProvider.js

```diff
@@ -0,0 +1,20 @@
+import React, { createContext, useContext, useReducer } from 'react';
+import {
+  func, objectOf, object, instanceOf,
+} from 'prop-types';
+
+export const StateContext = createContext();
+
+export const StateProvider = ({ reducer, initialState, children }) => (
+  <StateContext.Provider value={useReducer(reducer, initialState)}>
+    {children}
+  </StateContext.Provider>
+);
+
+export const useStateValue = () => useContext(StateContext);
```

Comment:
I dont see any use of these function, can you explain a little bit?

Reply:
> * missing errMessage, should render error component with request fail.

I don't think it will be a good idea to render error component at the home page.

Reply:
As I explained in my PR, StateProvider is a **general** **abstract** component that can be used to create [context](https://reactjs.org/docs/context.html#when-to-use-context). If any other components want to access the context provided by a StateProvider, it must use `useContext()` hook to access a "shared" state. For example. in your case, you need to access `page.offset` and `page.limit`. You can access it through `useStateValue` instead of writing  `import {useContext} from 'react'` and `useContext(context)`. I am just trying to prevent code smells here.

Reply:
thank you for explain